android/src/com/mapswithme/maps/routing/RoutingPlanController.java

```diff
@@ -188,26 +201,34 @@ private void updateProgressLabels()
                               RoutingController.formatArrivalTime(rinfo.totalTimeInSeconds)));
   }
 
-  public void updateBuildProgress(int progress, int router)
+  public void updateBuildProgress(int progress, @Framework.RouterType int router)
   {
     updateProgressLabels();
-
-    boolean vehicle = (router == Framework.ROUTER_TYPE_VEHICLE);
-    mRouterTypes.check(vehicle ? R.id.vehicle : R.id.pedestrian);
-
-    if (!RoutingController.get().isBuilding())
+    UiUtils.invisible(mProgressVehicle, mProgressPedestrian, mProgressBicycle);
```

Comment:
Там в вёрстке три подряд прогресса, если прятать - сьедет в левый край видимое. Там и раньше был UiUtils.visibleIf.
> On May 25, 2016, at 8:05 PM, Alexander Marchuk <notifications@github.com> wrote:
> 
> In android/src/com/mapswithme/maps/routing/RoutingPlanController.java <https://github.com/mapsme/omim/pull/3340#discussion_r64610491>:
> 
> >    {
> >      updateProgressLabels();
> > -
> > -    boolean vehicle = (router == Framework.ROUTER_TYPE_VEHICLE);
> > -    mRouterTypes.check(vehicle ? R.id.vehicle : R.id.pedestrian);
> > -
> > -    if (!RoutingController.get().isBuilding())
> > +    UiUtils.invisible(mProgressVehicle, mProgressPedestrian, mProgressBicycle);
> Почему не hide()?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/3340/files/0eaf8797da6003592602a2144d0bd41203a7c93b..fb35b52ab26a99751d3582d9e6e9c073e8d3b3fe#r64610491>